tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -310,6 +321,29 @@ EOF
 			return 1
 		fi
 
+		# A secure boot image for IBM SE should be rebuilt according to the KBS configuration.
+		if [ "${KATA_HYPERVISOR}" == "qemu-se" ]; then
+			if [ -z "${IBM_SE_CREDS_DIR:-}" ]; then
+				>&2 echo "ERROR: IBM_SE_CREDS_DIR is empty"
+				return 1
+			fi
+			creds_dir="${IBM_SE_CREDS_DIR}"
+			kernel_params_value="agent.guest_components_rest_api=resource agent.aa_kbc_params=cc_kbc::${svc_host}"
```

Comment:
Hi @BbolroC !

I started reading this commit with a feeling of "it seems to belong to somewhere else". Then I hit that line, which made me more inclined towards suggesting to wrap in a function to be called from the test. Why? Because if this `kernel_params_value` should match the one it's passed via Pod annotation, then it might be different from test to test.

For example, the recent added [test for sealed secrets](https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/k8s-sealed-secret.bats#L36) will pass the following kernel parameters instead: `agent.guest_components_procs=confidential-data-hub  agent.aa_kbc_params=cc_kbc::${svc_host}`

or am I missing something?

Reply:
Yes, from the beginning, this was wrapped as a function and called from k8s-confidential-attestation.bats. But, at the time of refactoring, I realised that file is the only one calling the function. That is why I had this snippet called once when the KBS is deployed. 

I totally agree with your opinion. I will change the code.